### DISCUSSION TITLE
Possible to customize the amount of milk foam?

### DISCUSSION THREAD

**Discussion by f2103** [TillFleisch/ESPHome-Philips-Smart-Coffee#78] [Q&A]
Hi guys,

I am planning on buying the EP223 (Series 2200) and install this mod. The device originally only offers Cappuccino which is filled with milk foam. As far as I know, it's not possible to manually steam milk without chosing an coffee beverage with this machine, means no other drinks with milk foam are possible.

I saw there are quite some customization possibilities inside ESPhome/HomeAssistant for bean amount etc.

Can I customize the amount of milk foam by using this mod? Or can I manually trigger steaming the milk and mix it myself?

Thx in advance! :)

**Comment by quack3d** [comment]
The manual approach to milk only on my machine (EP3243) is to choose cappuccino but also coffee grounds instead of beans. Then it starts with the milk and when done you have to hit the stop button before the water comes. I had a plan to automate that in HA, but haven't tried yet.

**Comment by quack3d** [comment]
Today I tried automating it via HA. The automation was set up to choose Cappuccino, ground coffee and milk size 1. After start I had it wait 30 seconds before pressing Start/Pause again for it to cancel the water from the coffee spout. The problem is that the time it takes to heat up before milk starts to pour is not the same each time. Maybe @TillFleisch can think of something clever. 😊

**Comment by TillFleisch** [comment]
> Can I customize the amount of milk foam by using this mod? Or can I manually trigger steaming the milk and mix it myself?

With this modification you can only do things remotely, which can be done using the display panel. Adjusting the milk size or only preparing foamed milk is likely not possible.

@quack3d's workaround may be a solution to your problem. Adjustment of the milk size could be achieved by early stopping.
I would recommend implementing the automation on the ESP itself using ESPHomes Scripts&Automation (see documentation).
As for the heating time mentioned by @quack3d, using the coffee machines state to "start the timer" would be a good idea. On my EP2220, when using the steam option, the staus sensor shows a `preparing` state before it actually starts spewing out steam. The duration of this state is different when the steam option has been used prior. Since the state sensor can be a bit slow to respond at times, this is also not an ideal solution, but the best I can come up with on the spot :/

**Reply by quack3d** [reply]
Unfortunately, when using the EP3243 I have to choose cappuccino or latte macchiato for it to put out steam. So it will say 'Brewing Cappuccino' from the moment I press start until the set milk amount is done and then start pouring in coffee from the grounds slot. So unless there's a hidden command to make it spew out steam only, making milk only on the 3000 series doesn't look easy.
